### PR TITLE
add pull request template

### PR DESCRIPTION
**Pull request by briandowns**
Proposed changes
======

Adds a template for pull request submitters to fill out providing additional insights into the change, and linking to a related issue.

Types of changes
======

New feature for pull requests. 

Verification
======

This PR is in the specified format. Once merged, this template will appear in a new pull request

Linked Issues
======
 

N/A

Further comments
======

N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+-->
+
+Linked Issues
+======
+<!--
+
+Link any related issues, pull-requests, or commit hashes that are relavent to this pull-request.
+
+If you are opening a PR without a corresponding issue create an issue before you do, [here](https://github.com/rancher/k3s/issues). This will help QA massively. PR's opened without linked issues will not be merged until an issue is created and linked here.
```
Comment (dweomer) [inline .github/PULL_REQUEST_TEMPLATE.md:35]:
```suggestion
If you are opening a PR without a corresponding issue please consider creating one first, [here](https://github.com/rancher/k3s/issues). This will help QA massively. PR's opened without linked issues will not be merged until an issue is created and linked here. 
```

**Comment by erikwilson** [comment]
Is there an issue associated with this PR?
(☞ﾟヮﾟ)☞

File: BUILDING.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ This repo includes all of Kubernetes history so `--depth 1` will avoid most of t
 
 To build the full release binary run `make` and that will create `./dist/artifacts/k3s`.
 
-Optionally to build the binaries using local go environment without running linting or building docker images:
+Optionally to build the binaries using local Go environment without running linting or building docker images:
```
Comment (erikwilson) [inline BUILDING.md:13]: Should this be a separate PR w/ linked issue?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+-->
+
+Linked Issues
+======
+<!--
+
+Link any related issues, pull-requests, or commit hashes that are relavent to this pull request.
+
+If you are opening a PR without a corresponding issue please consider creating one first, [here](https://github.com/rancher/k3s/issues). This will help QA massively. PR's opened without linked issues will not be merged until an issue is created and linked here.
```
Comment (erikwilson) [inline .github/PULL_REQUEST_TEMPLATE.md:35]:
```suggestion
If you are opening a PR without a corresponding issue please consider creating one first, [here](https://github.com/rancher/k3s/issues). A functional example will greatly help QA with verifying/reproducing a bug or testing new features.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<!-- HTML Comments can be left in place or removed, dealers choice. They are present simply to guide you on your pull-request journey. -->
```
Comment (erikwilson) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Question for anyone putting in PRs: Are HTML comments in the template a good idea?

**Comment by erikwilson** [comment]
Not trying to be silly, but do you mind putting this PR in the format of the template @briandowns? As you do so is there anything that you would like to see changed? Just trying to get a feel/example for how the template might look.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (brandond) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: I'm just a lowly community member, but imo the comments make more sense to me for issues where people might need to be guided through filling out the template. Hopefully anyone putting in a PR can figure it out on their own?

**Comment by briandowns** [comment]
> Not trying to be silly, but do you mind putting this PR in the format of the template @briandowns? As you do so is there anything that you would like to see changed? Just trying to get a feel/example for how the template might look.

Updated. Thanks for the comments!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (erikwilson) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: You would hope so, but for some where english is a second language or translated the comments may not be direct enough. I have also mistakenly entered information in a comment only to find out later it did not render correctly.

I like the approach longhorn has taken with issue templates a little more where it should be pretty obvious what needs to be filled in (https://raw.githubusercontent.com/longhorn/longhorn/master/.github/ISSUE_TEMPLATE/bug_report.md)

Ultimately I am ok with either approach tho, just wanted to open it to discussion.

Comment (briandowns) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Would it make sense to do a trial run of both for a period of time, measure their success or failures, and go with the one that makes things best, at the end of that trial period?
